core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -79,32 +108,32 @@ export class ExpressionSyntaxTreeService {
   constructor(private expressionParserService: ExpressionParserService) {}
 
   public getParamsUsedInExpression(expression: string): Array<string> {
-    let _findParams = (parseTree: Array<string>|string): Array<string> => {
-      let paramsFound = [];
-      if (parseTree instanceof Array) {
-        if (parseTree[0] === '#') {
-          paramsFound.push(parseTree[1]);
-        } else {
-          for (let i = 1; i < parseTree.length; i++) {
-            paramsFound = paramsFound.concat(_findParams(parseTree[i]));
-          }
-        }
-      }
+    let parsed = this.expressionParserService.parse(expression);
 
-      let uniqueParams = [];
-      for (let i = 0; i < paramsFound.length; i++) {
-        if (uniqueParams.indexOf(paramsFound[i]) === -1) {
-          uniqueParams.push(paramsFound[i]);
+    return this.findParams(parsed);
+  }
+
+  private findParams = (parseTree: Array<string>|string): Array<string> => {
+    let paramsFound = [];
+    if (parseTree instanceof Array) {
+      if (parseTree[0] === '#') {
+        paramsFound.push(parseTree[1]);
+      } else {
+        for (let i = 1; i < parseTree.length; i++) {
+          paramsFound = paramsFound.concat(this.findParams(parseTree[i]));
         }
       }
+    }
 
-      return uniqueParams.sort();
-    };
-
-    let parsed = this.expressionParserService.parse(expression);
+    let uniqueParams = [];
+    for (let i = 0; i < paramsFound.length; i++) {
+      if (uniqueParams.indexOf(paramsFound[i]) === -1) {
+        uniqueParams.push(paramsFound[i]);
+      }
+    }
 
-    return _findParams(parsed);
-  }
+    return uniqueParams.sort();
```

Comment:
Let's not sort here in this recursive function, it seems very wasteful. Let's sort the final array that gets returned by `getParamsUsedInExpression` instead.